### DISCUSSION TITLE
Roadmap: Autosave Draft Changes

### DISCUSSION THREAD

**Discussion by DanRibbens** [payloadcms/payload#295] [Roadmap]
### Description

In the admin UI when updating or creating a new document, the changes should be passed to the backend and persisted in draft form. This will prevent changes from being lost in the event the editor was working and disconnects before saving.

### Implementation Detail

The changes need to be persisted in draft form in an alternate collection to not interfere with live data. The admin UI will need to indicate the status of autosaves and could display a warning for connection interupts. This will need some server API endpoint to be enabled to acccept draft data.

In addition to autosaving data, Payload will need to feature some type of mechanism to "lock" documents when they are currently being edited by other users, in order to prevent users from losing their changes while editing a document at the same time.

### Potential Breaking Changes

This feature will need to be coordinated to not impact the Revision History (https://github.com/payloadcms/payload/discussions/292). We will also have to manage file uploads here too.

### Effort

Medium

**Comment by richardvanbergen** [comment]
I think a very important feature overall. At the moment I've just got a checkbox that says if something is published or not. This is much cleaner.

**Comment by realmacaw** [comment]
How much draft data are we talking about here? Because if data is less maybe Cookies can be considered to store drafts in browser? @DanRibbens @richardvanbergen What do you think about it 🤔?

**Reply by richardvanbergen** [reply]
No I don't think so because you can have multiple editors working on a draft before it's published. The way that sanity does it is:

1. Start editing
2. It autosaves at regular intervals on the server
3. There's a publish button where the draft becomes the document
4. You can add hooks to the publish action

**Comment by denolfe** [comment]
Released in [v0.15.0](https://github.com/payloadcms/payload/releases/tag/v0.15.0)
